bpm/bonita-api/bonita-common-api-impl/src/test/java/org/bonitasoft/engine/bpm/process/impl/ProcessDefinitionBuilderTest.java

```diff
@@ -489,4 +493,48 @@ public void userTask_with_null_constraint_condition_of_contract_should_throw_exc
         builder.done();
     }
 
+    @Test
+    public void validate_contract_with_no_type() throws Exception {
+        final ProcessDefinitionBuilder builder = new ProcessDefinitionBuilder().createNewInstance("test", "1");
+
+        builder.addContract().addInput("name", "desc");
+
+        expectedException.expect(InvalidProcessDefinitionException.class);
+        expectedException.expectMessage("Type not set on the contract input <name> of contract of the process");
+        builder.done();
+    }
+
+    @Test
+    public void validate_contract_with_no_type_on_user_task() throws Exception {
+        final ProcessDefinitionBuilder builder = new ProcessDefinitionBuilder().createNewInstance("test", "1");
+
+        builder.addActor("john").addUserTask("step1","john").addContract().addInput("name", "desc");
+
+        expectedException.expect(InvalidProcessDefinitionException.class);
+        expectedException.expectMessage("Type not set on the contract input <name> of contract of user task <step1>");
+        builder.done();
+    }
+
+    @Test
+    public void validate_contract_with_no_type_sub_children() throws Exception {
+        final ProcessDefinitionBuilder builder = new ProcessDefinitionBuilder().createNewInstance("test", "1");
+
+        builder.addContract().addInput("name", "desc").addChildren().addInput("name", "desc");
```

Comment:
Add why not have a method **addChildInput(String name, String description)** that you could call several times?